src/analytics/internal/analytics.js

```diff
@@ -0,0 +1,156 @@
+     * @param {notifType} type of notif event
+     */
+    async loadInititalData(notifType, isCacheUpdate = false) {
+        await db.transaction('rw', db.tables, async () => {
+            const eventLogCount = await db.eventLog
+                .where('type')
+                .anyOf(MapNotifTypeToIntArray[notifType])
+                .count()
+
+            const eventLogLastTimeUsed = await db.eventLog
+                .where('type')
+                .anyOf(MapNotifTypeToIntArray[notifType])
+                .last()
+
+            // If the data is out of date then we have to update with dexie query and initial loading of data
+            if (this._operationsMap[notifType] === undefined || isCacheUpdate) {
+                // if there is any entry in the dexie then update data of cache
+                if (eventLogLastTimeUsed) {
+                    this._operationsMap[notifType] = {
+                        last_time_used: eventLogLastTimeUsed.timestamp,
+                        value: eventLogCount,
+                    }
+                }
+            } else {
+                if (eventLogLastTimeUsed) {
+                    this._operationsMap[notifType] = {
+                        last_time_used: eventLogLastTimeUsed.timestamp,
+                        value:
+                            eventLogCount +
+                            this._operationsMap[notifType].value,
+                    }
+                }
+            }
+        })
+    }
+
+    incrementvalue(event) {
+        if (!this._operationsMap[event.notifType]) {
+            this._operationsMap[event.notifType] = {
+                value: 1,
+                last_time_used: event.last_time_used,
+            }
+        } else {
+            this._operationsMap[event.notifType] = {
+                value: this._operationsMap[event.notifType].value + 1,
+                last_time_used: event.last_time_used,
+            }
+        }
+    }
+
+    getCountNotif(notifType) {
+        return this._operationsMap[notifType]
+    }
+
+    /**
+     * Track any user-invoked events internally.
+     *
+     * @param {EventTrackInfo} eventArgs
+     */
+    async processEvent(eventArgs) {
+        this._loaded = 1
+        // Prepare the event to store the event in dexie db.
+        const timestamp = Date.now()
+
+        const params = {
+            ...eventArgs,
+            timestamp,
+        }
+
+        // If all the items has been loaded initially
+        if (this._loaded === 1) {
+            // If all the event has processed then directly save the event into db
+            if (!this._queueEvents.length) {
+                await this.storeEventLogStatistics(params)
+            } else {
+                while (this._queueEvents.length) {
+                    const event = this._queueEvents[0]
+                    this._queueEvents.shift()
+
+                    // Event happens on the client side
+                    if (event instanceof Object) {
```

Comment:
If I understand this right, the `registerOperations` method (which doesn't seem to get called from anywhere) will load up the `_queueEvents` state with `string` values (derived from the names of all tracked ops), while standard events pushed to the queue will be objects with `timestamp` and other data. Then here, when you get the next event from the queue state, you check the type of the event to decide on the course of action to take?

Even though JS can handle this with dynamic types, it's really making things way more complicated than they need to be. I wouldn't associate those `string` op names with an events queue - you could decide when to call `loadInitialData()` based on if the local state is empty or not when you go to update it. 
But thinking more about it, I don't really see the point of this queue state at all; let your storage abstraction deal with lots of incoming event store requests, and keep this `analytics` module as just a mediator between the storage + server modules and other Memex features.

Reply:
> the registerOperations method (which doesn't seem to get called from anywhere) 

This function will call when we will load the data initially (The data will load from dexie).
When we are loading data from dexie, in between any other can come, then we can push into queue and can execute when the operation loaded successfully.

Reply:
It's definitely not called from anywhere; there's 0 references.

Regardless of that, I still don't want `_queueEvents` state holding both object and string types, then having complicated type checks to perform completely different actions. But again, this internal analytics class doesn't need to have a queue state to push incoming event store requests to - leave this responsibility to storage (and only if needed for the storage backend; Dexie should handle this for you). 
All this class needs to do is keep track of its derived stats state (`_operationsMap`) and act as a mediator to forward event tracking requests from different extension features to both the storage and server.

You can easily determine if you need to call `loadInitialData` method when a new event comes in as the corresponding entry in the local `this._operationsMap` state will be `undefined` (does not matter if there's another running).

Reply:
If we don't want to use `_queueEvents`, we ca still manage things to get the right.
We can call directly get the data in all the memory variables by calling the `loadInitialData` directly and then we can handle all the incoming events without using queue. @poltak 
does it make sense?